### PR TITLE
dolphinEmu{Master}: removed unnecesary symlink, rename to dolphin-emu{-beta}

### LINKED ISSUE #145185: dolphinEmuMaster: creates two .desktop files
### Describe the bug
The derivation of `dolphinEmuMaster` creates a different .desktop file from the "official" one that has a different name. This is probably done in order to differentiate the two versions of `dolphinEmu` in case someone installs both of them, but the problem is that while it creates a new one, it doesn't delete the "official" one.
Also, the new .desktop file should also set `QT_QPA_PLATFORM=xcb`, otherwise games don't launch in Wayland. (Though maybe it should be a wrapper instead)
(And, unrelated, but why `dolphinEmu` and not `dolphin-emu`?)

The same also applies to PATH: there are both `dolphin-emu` and `dolphin-emu-master` (which is a link made by the derivation to `dolphin-emu` in the folder).

For consistency, there should be made a `dolphin-emu-nogui-master`, otherwise if both dolphins are installed, one overwrites the other in PATH.

### Steps To Reproduce
1. Install dolphinEmuMaster;
2. Open an app launcher;
3. There are both the original "Dolphin Emulator" and the new "Dolphin Emulator (master)" that open the same app (the master one doesn't set `QT_QPA_PLATFORM=xcb).
4. In PATH there are both `dolphin-emu` and `dolphin-emu-master` (which is a link).

### Expected behavior
There should only be the "Dolphin Emulator (master)" desktop file, and in PATH there should be only `dolphin-emulator-master` (and `dolphin-emu-nogui-master` instead of `dolphin-emu-nogui`).

### Notify maintainers

@MP2E @ashkitten

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Motivation for this change
Fixes #145185. Kinda, Now there's only a `dolphin-emu` executable and "Dolphin Emulator" .desktop file, regardless if you install `dolphin-emu` or `dolphin-emu-master`. (Other packages on nixpkgs with different versions don't have different executable names, example wine)

About the name change, consinstency. I don't know if it's fine to change a package name like this. Tell me if it's not.
EDIT: From `dolphin-emu-master` to `dolphin-emu-beta`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4571,8 +4571,8 @@ with pkgs;
 
   dotnetfx40 = callPackage ../development/libraries/dotnetfx40 { };
 
-  dolphinEmu = callPackage ../misc/emulators/dolphin-emu { };
-  dolphinEmuMaster = qt5.callPackage ../misc/emulators/dolphin-emu/master.nix {
```
Comment (thiagokokada) [inline pkgs/top-level/all-packages.nix:4575]: Please add aliases to the old names on `pkgs/top-level/aliases.nix` to avoid breaking old configurations.

**Comment by SuperSamus** [comment]
Not really sure if I should use the -master suffix, though.
The notation is present only in the downloaded files, there is no mention of "master" in the official [download page](https://dolphin-emu.org/download/). Maybe -dev is better?
Probably thinking it too much.

**Comment by ashkitten** [comment]
we've been updating the package for beta releases, so `dolphin-emu-beta` is more appropriate i think

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145384` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>dolphin-emu</li>
    <li>dolphin-emu-beta</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**